test/celeritas/optical/ScintillationGenerator.test.cc

```diff
@@ -0,0 +1,184 @@
+        input_.post_velocity = 0.9 * input_.pre_velocity;
+        input_.pre_pos = {0, 0, 0};
+        input_.post_pos = {0, 0, 1};
+        input_.charge = units::ElementaryCharge{-1};
+        input_.matId = OpticalMaterialId{0};
+    }
+    //! Get random number generator with clean counter
+    RandomEngine& rng()
+    {
+        rng_.reset_count();
+        return rng_;
+    }
+
+  protected:
+    // Host/device storage and reference
+    HostCRef<OpticalPropertyData> properties;
+    std::shared_ptr<ScintillationParams const> params;
+    OpticalMaterialId material{0};
+    units::ElementaryCharge charge{-1};
+
+    RandomEngine rng_;
+    ScintillationInput input_;
+};
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+TEST_F(ScintillationGeneratorTest, basic)
+{
+    // Output data
+    std::vector<OpticalPrimary> storage;
+    storage.reserve(input_.num_photons);
+
+    // Create the generator
+    ScintillationGenerator generate_photons(
+        input_, params->host_ref(), make_span(storage));
+    RandomEngine& rng_engine = this->rng();
+
+    // Generate optical photons for a given input
+    auto photons = generate_photons(rng_engine);
+
+    // Check results
+    std::vector<real_type> energy;
+    std::vector<real_type> time;
+    std::vector<real_type> cos_theta;
+
+    for (auto i : range(input_.num_photons))
+    {
+        energy.push_back(photons[i].energy.value());
+        time.push_back(photons[i].time);
+        cos_theta.push_back(dot_product(photons[i].direction,
+                                        input_.post_pos - input_.pre_pos));
+    }
+
+    const real_type expected_energy[] = {9.35613547878811e-06,
+                                         9.39574581587642e-06,
+                                         1.09240249982534e-05,
+                                         9.10710182050691e-06};
+
+    const real_type expected_time[] = {7.30250028666843e-09,
+                                       1.05142594015847e-08,
+                                       1.24626439432502e-08,
+                                       2.11861667895083e-09};
+
+    const real_type expected_cos_theta[] = {0.937735542248463,
+                                            -0.77507096788764,
+                                            0.744857640134601,
+                                            -0.182537678076479};
+
+    EXPECT_VEC_SOFT_EQ(expected_energy, energy);
+    EXPECT_VEC_SOFT_EQ(expected_time, time);
+    EXPECT_VEC_SOFT_EQ(expected_cos_theta, cos_theta);
+}
+
+//---------------------------------------------------------------------------//
+
+TEST_F(ScintillationGeneratorTest, stress_test)
+{
+    // Generate a large number of optical photons
+    input_.num_photons = 1e+6;
+
+    // Output data
+    std::vector<OpticalPrimary> storage;
+    storage.reserve(input_.num_photons);
```

Comment:
Not sure if you tried running with debug assertions on, but I'm seeing some failures in the tests, e.g. from:
```suggestion
    std::vector<OpticalPrimary> storage(input_.num_photons);
```

Reply:
Sorry that I did not turn on DEBUG! I think that this error is somewhat related to the use of OpticalPropertyData for ScintillationGenerator.test (in which refraction_index data is not built for the scintillation test.  I will address this with the rest of comments by @sethrj this afternoon.  Many thanks!!

Reply:
True, probably we should not require that refractive index (or any optical properties) be assigned in `OpticalPropertyData`.